operator/docs/operations.md

```diff
@@ -1,3 +1,45 @@
+
+#### SAST Checking & Linters
+Pipeline Service uses Static Application Security Testing (SAST) to scan the codebase for potential security vulnerabilities. This is done using various tools such as SonarQube, Bandit, and other open-source tools. The SAST scans are run automatically as part of the Continuous Integration (CI) process, and results are reported back to the development team. 
+
+The below tools are currently used in Pipeline Service: 
+- [checkov](https://github.com/bridgecrewio/checkov)
+- [hadolint](https://github.com/hadolint/hadolint)
+- [shellcheck](https://github.com/koalaman/shellcheck)
+- [yamllint](https://github.com/adrienverge/yamllint)
+
+
+#### Vulnerability Scanning
+
+Pipeline Service also employs vulnerability scanning to detect and address potential security issues in its dependencies. This is done using Clair. The vulnerability scans are run regularly, and the results are analyzed by the team to identify any potential risks or issues.
+
+### Tests
+
+Pipeline Service employs a rigorous testing process to ensure the quality and reliability of the service. We run tests on every PR as part of the CI process to catch any issues early in the development cycle. As part of each PR, we run SAST checks to identify potential security vulnerabilities and functional tests to ensure that Pipeline Service remains intact.
+
+We use Tekton Pipelines to run our tests. All our tasks, pipelines, and pipelineruns live in the [.tekton folder](../../.tekton) in our repository. This approach allows us to define the test pipelines in a declarative way, making it easy to maintain and modify. In the future, we plan to dogfood Pipeline Service by using it to test itself. This approach also provides an opportunity to identify any potential issues with the service itself and address them before they affect our users.
+
+### Upgrade Strategy for Operators
+
+Pipeline Service uses Operators, specifically OpenShift Pipelines and OpenShift GitOps, to manage the deployment and configuration of its components.
+
+We use a versioned approach in upgrading Pipeline Service. We control the version of the components that are deployed and managed by the OpenShift Pipelines operator by specifying the source and channel in the [Subscription](../../operator/gitops/argocd/pipeline-service/openshift-pipelines/openshift-operator.yaml) of the operator. Using this approach, we can upgrade the components of Pipeline Service in a consistent and reliable way.
```

Comment:
>We use a versioned approach in upgrading Pipeline Service

I'd rather say we use a gitops approach to upgrade pipeline-service (or a git versioned approach) to highlight that git is our source of truth for the deployed config.

Reply:
Done.